### PR TITLE
Get the mountInfo by 'LookupMount' in containerd

### PR DESCRIPTION
**Pull request by miaoyq**
Signed-off-by: Yanqiang Miao <miao.yanqiang@zte.com.cn>

Address [this comment](https://github.com/kubernetes-incubator/cri-containerd/pull/257#issuecomment-330152858).  
/cc @Random-Liu

### CODE REVIEW

**Comment by miaoyq** [comment]
There is something wrong in my local environment, `make test-cri` always failed, see #298 
So I have not executed `make test-cri` locally that cause the CI failed, I will debug this soon.

File: pkg/server/helpers.go [hunk 1]
```diff
@@ -405,27 +405,17 @@ func isInCRIMounts(dst string, mounts []*runtime.Mount) bool {
 	return false
 }
 
-//TODO: Replace with `mount.Lookup()`in containerd after #257 is marged
```
Comment (Random-Liu) [inline pkg/server/helpers.go]: Why do we still need this loop?
`LookupMount` checks device id directly: https://github.com/containerd/containerd/blob/master/mount/lookup_unix.go

File: pkg/server/helpers.go [hunk 2]
```diff
@@ -405,27 +405,17 @@ func isInCRIMounts(dst string, mounts []*runtime.Mount) bool {
...
-	mountinfo := getMountInfo(mountInfos, source)
-	if mountinfo != nil {
+func getSourceMount(source string, lookupMount func(string) (mount.Info, error)) (string, string, error) {
+	mountinfo, err := lookupMount(source)
+	if err == nil {
 		return source, mountinfo.Optional, nil
 	}
 
 	path := source
 	for {
```
Comment (miaoyq) [inline pkg/server/helpers.go]: @Random-Liu Do you mean this loop?

**Comment by Random-Liu** [comment]
Yeah

File: pkg/server/helpers.go (hunk 2 above)
Comment (Random-Liu) [reply pkg/server/helpers.go]: Yeah

Comment (miaoyq) [reply pkg/server/helpers.go]: Seems not necessary, I will remove this loop, Thanks :-)

**Comment by miaoyq** [comment]
@Random-Liu  Have updated the code based on #342, PTAL

File: pkg/server/container_create.go [hunk 3]
```diff
@@ -818,8 +814,8 @@ func generateApparmorSpecOpts(apparmorProf string, privileged, apparmorEnabled b
 }
 
 // Ensure mount point on which path is mounted, is shared.
-func ensureShared(path string, mountInfos []*mount.Info) error {
-	sourceMount, optionalOpts, err := getSourceMount(path, mountInfos)
+func ensureShared(path string, lookupMount func(string) (mount.Info, error)) error {
+	sourceMount, optionalOpts, err := getSourceMount(path, lookupMount)
```
Comment (Random-Liu) [inline pkg/server/container_create.go]: Do we still need `getSourceMount`? I think we could just call `lookupMount` directly here.

File: pkg/server/container_create_test.go [hunk 4]
```diff
@@ -24,7 +24,8 @@ import (
 	"github.com/containerd/containerd"
 	"github.com/containerd/containerd/contrib/apparmor"
 	"github.com/containerd/containerd/contrib/seccomp"
-	"github.com/docker/docker/pkg/mount"
+	//"github.com/docker/docker/pkg/mount"
```
Comment (Random-Liu) [inline pkg/server/container_create_test.go]: Remove the comment.

**Comment by Random-Liu** [comment]
LGTM overall other than the comments.

File: pkg/server/container_create.go (hunk 3 above)
Comment (miaoyq) [reply pkg/server/container_create.go]: Done.

File: pkg/server/container_create_test.go (hunk 4 above)
Comment (miaoyq) [reply pkg/server/container_create_test.go]: Done.

**Pull request merged**